R/apa_print_anova.R

```diff
@@ -61,66 +74,180 @@ apa_print.summary.aov <- function(x, ...) {
   }
 
-  apa_print(summary_x, correction = correction, intercept = intercept, ...)
+  apa_print(summary_x, correction = correction, intercept = intercept, .x = x, ...)
 }
 
 
 #' @rdname apa_print.aov
 #' @method apa_print summary.Anova.mlm
 #' @export
 
-apa_print.summary.Anova.mlm <- function(x, correction = getOption("papaja.sphericity_correction"), intercept = FALSE, ...) {
-  variance_table <- arrange_anova(x, correction)
+apa_print.summary.Anova.mlm <- function(
+  x
+  , correction = getOption("papaja.sphericity_correction")
+  , intercept = FALSE
+  , estimate = getOption("papaja.estimate_anova", "ges")
+  , mse = getOption("papaja.mse")
+  , observed = NULL
+  , in_paren = FALSE
+  , ...
+) {
+
+  intercept <- isTRUE(intercept)
+  in_paren  <- isTRUE(in_paren)
+
+  arranged_table <- arrange_anova(x, correction) # arrange_anova.summary.Anova.mlm
+
+  ellipsis <- list(...)
+  .x <- ellipsis$.x
+  ellipsis$.x <- NULL
 
-  print_anova(variance_table, intercept = intercept, ...)
+  canonical_table <- canonize(arranged_table)
+  canonical_table <- add_custom_effect_sizes(
+    canonical_table
+    , estimate = estimate
+    , mse = mse
+    , observed = observed
+    , intercept = intercept
+    , .x = .x
+  )
+
+  # Remove intercept if the user doesn't want it:
+  if(!intercept) canonical_table <- canonical_table[canonical_table$term != "(Intercept)", , drop = FALSE]
+
+  ellipsis <- defaults(
+    ellipsis
+    , set.if.null = list(
+      digits = 3L
+      , gt1 = FALSE
+    )
+  )
+  ellipsis$x <- canonical_table
+  beautiful_table <- do.call("beautify", ellipsis)
+
+
+  glue_apa_results(
+    beautiful_table
+    , est_glue = construct_glue(beautiful_table, "estimate")
+    , stat_glue = construct_glue(beautiful_table, "statistic")
+    , term_names = sanitize_terms(unlabel(canonical_table$term))
+    , in_paren = in_paren
+    , est_first = FALSE
+    , simplify = FALSE
+  )
 }
 
 
+
+
 #' @rdname apa_print.aov
 #' @method apa_print afex_aov
 #' @export
 
-apa_print.afex_aov <- function(x, correction = getOption("papaja.sphericity_correction"), intercept = FALSE, ...) {
+apa_print.afex_aov <- function(
+  x
+  , correction = getOption("papaja.sphericity_correction")
+  , intercept = FALSE
+  , ...
+) {
   validate(intercept, check_class = "logical", check_length = 1)
 
   afex_aov_intercept <- "(Intercept)" %in% rownames(x$anova_table)
   if(afex_aov_intercept != intercept & afex_aov_intercept) {
     warning("In your call of afex::aov_car() you requested the intercept term, but now you did not (in apa_print 'intercept = FALSE' is the default). Thus, the intercept term will be omitted; make sure this is what you want.")
   }
 
-  if("Anova.mlm" %in% class(x$Anova)) {
+  if(inherits(x$Anova, "Anova.mlm")) {
     summary_x <- summary(x$Anova)
-    apa_print(summary_x, correction = correction, intercept = intercept, ...)
+    apa_print(summary_x, correction = correction, intercept = intercept, .x = x, ...) # apa_print.summary.Anova.mlm
```

Comment:
If we want to carry the `observed` setting from `afex` forward, we could check for the specification of `observed` in the ellipsis and if it is `NULL` default `observed` to this:

~~~r
attr(x$anova_table, "observed") 
~~~

This yields `character(0)` if no observed variables have been specified. Any thoughts?

Reply:
I'm already working on this

Reply:
Should we really allow the user to specify `observed` differently than what is specified in the model? I'm inclined to overwrite the user's argument (passed to `apa_print()`), or at least warn if there is a disparity

Reply:
I just committed a change that will give an informative warning when `observed` and the information from the `afex_aov` object differ.